crates/ruff_db/src/vendored.rs

```diff
@@ -273,12 +299,18 @@ fn normalize_vendored_path(path: &VendoredPath) -> NormalizedVendoredPath {
             camino::Utf8Component::Normal(part) => normalized_parts.push(part),
             camino::Utf8Component::CurDir => continue,
             camino::Utf8Component::ParentDir => {
-                normalized_parts.pop();
+                if normalized_parts.pop().is_none() {
+                    return Err(InvalidVendoredPathError::EscapeFromZipFile);
+                }
```

Comment:
I quickly checked what the normal FS operations return when you navigate outside the `cwd`. They just return a `NotFound` error, which is probably also enough for us.